### DISCUSSION TITLE
Partage des référentiels

### DISCUSSION THREAD

**Discussion by jpose** [dbarzin/deming#45] [Q&A] [answered]
Bonjour à tous, avez-vous déjà intégré d'autres référentiels de sécurité (PCIDSS, BIM, NIS, CIS etc ..) dans Deming ? Si oui est-il possible de partager les fichiers pouvant être intégré dans Deming ? Merci 👍

**Comment by fgilain** [comment]
sans oublier DORA  ;-))

**Reply by jpose** [reply]
@dbarzin au-delà des indications d'import dans Deming, avez-vous un fichier de template d'intégration d'ISO 27001:2022 qui pourrait être plus pertinent à comprendre et qui permettrait de créer rapidement ces fichiers ? Merci 👍

**Reply by dbarzin** [reply]
J’accepte les contributions !

**Comment by dbarzin** [comment]
Deming est fournit en standard avec les mesures de sécurité de la norme ISO 27001:2022.
Pour peupler la base de données avec ces mesures, il faut exécuter les commandes suivantes :

    php artisan db:seed --class=AttributeSeeder
    php artisan db:seed --class=DomainSeeder
    php artisan db:seed --class=MeasureSeeder

C'est décrit dans la procédure [d'installation.](https://github.com/dbarzin/deming/blob/main/INSTALL.fr.md).

**Reply by fgilain** [reply]
Bonjour,

Celles ci sont semble t'il en Francais, existent t'elles en Anglais ?

Merci

FG

**Reply by jpose** [reply]
Personnellement je peux contribuer à créer les fichiers Excel pour intégration. La procédure d'installation fait référence à une intégration en direct dans la base de données. J'ai lu que lors de la mise à jour permettant une intégration par import de fichier, vous aviez créer un fichier de test. Je voudrais juste avoir une vue sur ce fichier pour le reproduire à l'identique mais avec d'autres référentiels. Merci 👍

**Reply by fgilain** [reply]
je compte aussi intégrer des référentiels comme Swift CSCF.

**Reply by dbarzin** [reply]
Il existe une version anglaise des contrôles ISO 27001:2022. Elles se sont perdues lors d'une modification de base de données pour Docker. Elles vont être disponibles rapidement.
Le plus simple est de passer par des fichiers Excel via le menu Configuration->Import

**Reply by dbarzin** [reply]
C'est disponible dans le répertoire "referentials".
Il y a la 27001:2022 en français et anglais ansi que la  [NIST SP 800-53](https://www.nist.gov/privacy-framework/nist-sp-800-53).

**Reply by fgilain** [reply]
Sauf erreur de ma part, le contenu du fichier intitulé EN est aussi en FR, non ?

**Reply by dbarzin** [reply]
Excellent, merci, c'est corrigé !

**Reply by jpose** [reply]
Super, je vais m'y mettre pour intégrer PCIDSS dans un 1er temps merci !

**Comment by JinxCH73** [comment]
Attention quand même, en tout cas pour ISO, on est supposé acheter la norme, qui n'est pas disponible gratuitement.

**Reply by dbarzin** [reply]
Ne sont repris que les objectifs de contrôle que l'on retrouve partout sur internet, les métriques sont un interprétation propre à Deming qui doivent être adaptées pour chaque organisation.

**Comment by LoLoow** [comment]
Y a t-il le référentiel 27001:2013 disponible quelque part importable à la manière du référentiel de 2022 ?
En vous remerciant !

**Comment by dbarzin** [comment]
Pour importer le référentiel ISO 27001:2013, repartir d'une base de données vide :

    php artisan migrate:fresh --seed

puis importer le référentiel avec la commande

    sudo mysql deming < deming-27001_2013.sql

Pour générer une base de données de test

    php artisan deming:generateTests

**Comment by danje57** [comment]
J'ai fait le référentiel PCI DSS v4.0 pour Monarc.

https://objects.monarc.lu/object/view/5238

Si j'ai bien compris, pour Deming, il faudrait créer les domaines à la main dans Deming, puis importer les controles via la fonction import, qui linkera par la suite l'ensemble des contrôles aux domaines créés?

Ou alors, l'import du fichier Excel permettra aussi de créer les domaines?

Si c'est que ça, je pourrai faire le référentiel PCI DSS sur base de ce que j'ai fait pour Monarc.

**Reply by dbarzin** [reply]
Exactement, merci pour la contribution !

**Comment by dbarzin** [comment]
L'import du fichier Excel permet aussi de créer les domaines.

**Reply by danje57** [reply]
Ok Merci! Je vais donc le faire sur base de ce que j'avais fait pour Monarc.

**Reply by danje57** [reply]
Je viens d'importer le référentiel PCI DSS v4.0 EN:

![image](https://github.com/dbarzin/deming/assets/16591723/320644ae-efa2-4d3e-8d5b-92bb969da09d)

Les domaines sont biens créés:

![image](https://github.com/dbarzin/deming/assets/16591723/d5bfa929-3c6a-4444-91fd-dd472164e883)

La question maintenant, c'est est-ce que je créé 1 ligne par sous controle de PCI DSS?
ça fait 281 prérequis de la norma. Soit un paquet de ligne à créer...

**Reply by dbarzin** [reply]
C'est exactement cela...

**Reply by danje57** [reply]
Voilà donc:

Avec une idée d'amélioration :), peux-tu ajouter dans le fichier Excel, la possibilité de remplir le champ description du nom de domaine? Après l'import, il est nécessaire de le remplir à la main.

Voici donc ce que ça donne:

280 Contrôles.

![image](https://github.com/dbarzin/deming/assets/16591723/93c022c0-1da7-4c8a-89f1-21225be19a0c)

![image](https://github.com/dbarzin/deming/assets/16591723/ee7f6d25-89d7-4b3a-b8bf-1fa66fc8c073)

![image](https://github.com/dbarzin/deming/assets/16591723/12fdfca9-40c1-4c23-8d6a-d78fdb1c00a0)

![image](https://github.com/dbarzin/deming/assets/16591723/51a53798-2bf8-43eb-bea4-773bc18e53c1)

![image](https://github.com/dbarzin/deming/assets/16591723/c1db995c-3488-4936-893a-4f332a998a62)


Dis moi si ça te va.

**Reply by danje57** [reply, marked as answer]
[PCI DSS 4.0 EN b001.xlsx](https://github.com/dbarzin/deming/files/15285604/PCI.DSS.4.0.EN.b001.xlsx)

**Reply by dbarzin** [reply]
Il est maintenant possible de fournir la description du domaine dans le fichier d'import.

**Reply by danje57** [reply]
Je mets just à jour Deming, et je dispose de la fonctionalité?

Il suffit juste d'ajouter un champ description dans le fichier Excel? Mais il y'a déjà un Champ description  :p

**Reply by dbarzin** [reply]
Oui, `git pull`. La documentation sur l'écran d'import a été mis à jour. Le champ description du domaine se trouve juste après la colonne du titre du domaine.

**Reply by danje57** [reply]
Voilà je viens de mettre à jour.

[PCI DSS 4.0 EN b002.xlsx](https://github.com/dbarzin/deming/files/15298579/PCI.DSS.4.0.EN.b002.xlsx)

Petite idée de suggestion GUI, agrandir la combobox en fonction de la taille du plus grand string.

![image](https://github.com/dbarzin/deming/assets/16591723/a9064367-2dfd-4709-b3ef-059c5479c20b)

**Reply by dbarzin** [reply]
OK, la taille de ce champ a été élargi.

**Comment by dbarzin** [comment]
L'écran d'importation a été revu, il est maintenant possible d'importer un référentiel depuis un repository (répertoire storage/app/repository), d'effacer la base de données et de générer des données de test :

![image](https://github.com/dbarzin/deming/assets/43208232/f3acc23f-a01a-44aa-b2be-cf40fcf8b002)
